the-collab-lab/tcl-30-whats-near-me
# [TCL30-9, TCL30-5]  Fetch data from Wikipedia 'nearby' API and render it on the map

## Description
This PR adds ``Spinner``, ``ErrorMessage``, ``WikiResults`` components to the app. Along with that, it fetches Paris locations with Wikipedia entries from the ‘nearby’ API proxy endpoint and makes the data returned available to the ``<Map/>`` component. This PR also adds custom pins to the map, using data fetched from "nearby" API

## Related Issue

Closes [TCL30-5](https://the-collab-lab.atlassian.net/browse/TCL30-5) and [TCL30-9](https://the-collab-lab.atlassian.net/browse/TCL30-9)


## Acceptance Criteria

- When the app starts, a call should be made to the proxy endpoint using the latitude and longitude at the centre of the map

- The returned data should be saved to state or in some other way made available to rendering components 

- There are locations (e.g., the middle of the ocean) where we will not get any locations back from the API. The app should handle that case without generating errors

- Data fetched from "nearby" API should be used to render pins on the map

- Pins should be 24px x 24px in size and should have a little down caret on the bottom edge of the pin pointing to the location of the place

- Pins should show a scaled-down thumbnail image in cases when a thumbnail image is present

- Pins should render a default pin image when no thumbnail is returned

- Pins should flow in onto the map

## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|     | :bug: Bug fix              |
| ✓   | :sparkles: New feature     |
|     | :hammer: Refactoring       |
|     | :100: Add tests            |
|     | :link: Update dependencies |
|     | :scroll: Docs              |

## Updates

### Before

There was no access to the Wikipedia 'nearby' entries in the app
![before_upd](https://user-images.githubusercontent.com/16920088/126311118-b0ea9465-b3ff-424f-b15c-a7bc0f97532f.png)

### After

App components have access to the data fetched from the Wikipedia 'nearby' API. The returned data is used to render custom pins on the map
![after555](https://user-images.githubusercontent.com/16920088/126359298-940b8d21-d515-414b-965f-0e4f4b128e60.png)



## Testing Steps / QA Criteria

- From your terminal, pull down this branch - ``ku-mm-fetch-wiki-data``
-  Run ``npm start`` to launch the app
-  Inspect state for the expected data
![devtools](https://user-images.githubusercontent.com/16920088/126078476-e321e5cf-9bd3-42e3-b9f6-570ef43fa335.png)
-  Launch the app with a location that has been verified not to return any Wikipedia entries. For example,  replace ``const { lat, lng } = { lat: 48.8566, lng: 2.3522 }``  in the WikiResults component with `` const { lat, lng } = { lat: 8.7832, lng: 124.5085 }``
-  Make sure that for the locations verified to return Wikipedia entries, pins are rendered on the map
-  Make sure those pins show a scaled-down thumbnail image in cases when a thumbnail image is present and their size is 24x24
-  Make sure the pins render a default pin image when no thumbnail is returned
-  Make sure the pins flow in onto the map
